docs/faq.rst

```diff
@@ -37,6 +37,8 @@ Type                                Status
 HDF5 "time" type
 NumPy "U" strings                   No HDF5 equivalent
 NumPy generic "O"                   Not planned
+Any other NumPy dtype               Map to opaque type with specific tag, You
+                                    must register types manually
```

Comment:
I'd like this to cross-reference `register_dtype` for people to find more information.